security-utils/src/main/java/com/yahoo/security/tls/ConnectionAuthContext.java

```diff
@@ -7,28 +7,77 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.Set;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
 
 import static com.yahoo.security.SubjectAlternativeName.Type.DNS;
 import static com.yahoo.security.SubjectAlternativeName.Type.URI;
+import static com.yahoo.security.tls.CapabilityMode.DISABLE;
+import static com.yahoo.security.tls.CapabilityMode.LOG_ONLY;
 
 /**
  * @author bjorncs
  */
 public record ConnectionAuthContext(List<X509Certificate> peerCertificateChain,
                                     CapabilitySet capabilities,
-                                    Set<String> matchedPolicies) {
+                                    Set<String> matchedPolicies,
+                                    CapabilityMode capabilityMode) {
 
-    private static final ConnectionAuthContext DEFAULT_ALL_CAPABILITIES = new ConnectionAuthContext(List.of());
+    private static final Logger log = Logger.getLogger(ConnectionAuthContext.class.getName());
 
     public ConnectionAuthContext {
         peerCertificateChain = List.copyOf(peerCertificateChain);
         matchedPolicies = Set.copyOf(matchedPolicies);
     }
 
-    private ConnectionAuthContext(List<X509Certificate> certs) { this(certs, CapabilitySet.all(), Set.of()); }
+    private ConnectionAuthContext(List<X509Certificate> certs, CapabilityMode capabilityMode) {
+        this(certs, CapabilitySet.all(), Set.of(), capabilityMode);
+    }
 
     public boolean authorized() { return !capabilities.hasNone(); }
 
+    public boolean hasCapabilities(CapabilitySet requiredCapabilities) {
+        return hasCapabilities(requiredCapabilities, null, null, null);
+    }
+
+    /** Provided strings are used for improved logging only */
+    public boolean hasCapabilities(CapabilitySet requiredCapabilities, String action, String resource, String peer) {
```

Comment:
`hasCapabilities` sounds like a very pure and innocent function with no side effects. Maybe it should be called `checkCapabilities` or something similar? Though that in turn might sound like an exception is thrown on failure... Consider factoring out the actual mode checking and logging once we see how the capability checking piece fits into the grander authz puzzle.